src/DataViews/LiveFunctionsDataView.cpp

```diff
@@ -148,22 +147,22 @@ std::vector<uint64_t> LiveFunctionsDataView::GetFunctionTimerDurations(int row)
                    return timer->end() - timer->start();
                  });
 
+  std::sort(timer_durations.begin(), timer_durations.end());
   return timer_durations;
 }
 
 void LiveFunctionsDataView::OnSelect(const std::vector<int>& rows) {
   UpdateHighlightedFunctionId(rows);
   UpdateSelectedFunctionId();
 
-  std::vector<uint64_t> timer_durations;
+  std::vector<uint64_t>* timer_durations = nullptr;
   std::string function_name;
   if (!rows.empty()) {
     const FunctionInfo& function = *GetFunctionInfoFromRow(rows[0]);
     function_name = orbit_client_data::function_utils::GetDisplayName(function);
-
-    timer_durations = GetFunctionTimerDurations(rows[0]);
+    timer_durations = &timer_durations_.at(indices_[rows[0]]);
   }
-  app_->ShowHistogram(std::move(timer_durations), function_name);
+  app_->ShowHistogram(timer_durations, function_name);
```

Comment:
Nit: prefer early return here.
```
if (rows.empty()) return;

std::string function _name = orbit_client_data::function_utils::GetDisplayName(function);
std::vector<uint64_t>* timer_durations = &timer_durations_.at(indices_[rows[0]]);
app_->ShowHistogram(timer_durations, function_name);

```

Reply:
We need to call `app_->ShowHistogram` either way. If nothing is selected, we call it with `nullptr` and other defaults to remove the histogram from the widget.

Reply:
Thanks for the explanation! I didn't realise we need the call to remove the histogram. XD

Reply:
That's because the code is bad. I'we re-written it. Hope, it's clearer now.